### PR TITLE
AV-56761: Re-factor of go controller phase 1

### PR DESCRIPTION
**Pull request by sudswasavi**
This is a phased re-factoring effort of the present k8s go controller
code. This phase creates a utils package and moves some of the go
files to this package. Due to this movement, there's a huge re-factor
that was needed in the other files. One significant change is the log
file init.

### CODE REVIEW

**Comment by rangar** [comment]
@sudswasavi is it possible to keep indentation to 4 spaces ?

File: avi_k8s_controller/pkg/utils/utils.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+import (
+	"net"
+	"net/url"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	extensions "k8s.io/api/extensions/v1beta1"
+)
+
+func IsV4(addr string) bool {
```
Comment (adityats) [inline avi_k8s_controller/pkg/utils/utils.go]: let's use Go's inbuilt routines for these methods

**Comment by sudswasavi** [comment]
@rangar I ran a gofmt on the project that usually puts 8 spaces :/
I went around to verify this in standard go projects. Istio for example: https://github.com/istio/istio/blob/master/pilot/pkg/bootstrap/monitoring.go#L60

File: avi_k8s_controller/pkg/utils/utils.go (hunk 1 above)
Comment (sudswasavi) [reply avi_k8s_controller/pkg/utils/utils.go]: @adityats this seems like a standard way. https://golang.org/pkg/net/#IP.To4 I couldn't find any other off the bat library to do this.

**Comment by protobuf-reviewers-avinetworks** [comment]
@sudswasavi I believe the rest of our code incl. Go code is 4 spaces right ? We should follow the rest of Avi coding conventions

**Comment by sudswasavi** [comment]
@protobuf-reviewers-avinetworks : Not really: https://github.com/avinetworks/sdk/blob/master/go/examples/create_vs.go#L14

**Pull request merged**